lib/omnibus/cli.rb

```diff
@@ -85,8 +81,8 @@ def build(name)
       project = Project.load(name, manifest)
       say("Building #{project.name} #{project.build_version}...")
       # Not sure why we need the !Omnibus::S3Cache.fetch_missing.empty? check here
-      Omnibus::S3Cache.populate if @options[:populate_s3_cache] && !Omnibus::S3Cache.fetch_missing.empty?
-      Omnibus::S3LicenseCache.populate if @options[:populate_s3_cache]
+      Omnibus::S3Cache.populate if Omnibus::Config.use_s3_caching && Omnibus::Config.s3_authenticated_download && !Omnibus::S3Cache.fetch_missing.empty?
```

Comment:
Why  `Omnibus::Config.s3_authenticated_download`?

Reply:
I see; that's interesting, I hadn't thought that we might be using the cache like that. Which, by the way, does feel a bit brittle, does that mean that we have a hidden dependency from the macOS build to the rest of builds then?